### PR TITLE
Bug fixes for group language anddesignation validation

### PR DESCRIPTION
**Pull request by ravisaurav-tarento**

### CODE REVIEW

File: src/main/java/org/sunbird/workflow/config/Configuration.java [hunk 1]
```diff
@@ -804,4 +807,12 @@ public List<String> getBulkUploadCategoryValue() {
     public void setBulkUploadCategoryValue(String bulkUploadCategoryValue) {
         this.bulkUploadCategoryValue = bulkUploadCategoryValue;
     }
+
+    public String getGroupValues() {
+        return groupValues;
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/workflow/config/Configuration.java]: Just return this as List<String> -- refer the cb-ext implementation for the same for how comma separated String is returned as String.

File: src/main/java/org/sunbird/workflow/service/UserBulkUploadService.java [hunk 2]
```diff
@@ -629,4 +643,33 @@ private boolean validateCategory(String category) {
         return categoryValues != null && categoryValues.contains(category);
     }
 
+    private boolean validateGroupValue(String groupValue){
+        String groupValues = configuration.getGroupValues();
+        Set<String> groupValuesSet = Stream.of(groupValues.split(",")).collect(Collectors.toSet());
+        return groupValuesSet.contains(groupValue);
+    }
+
+    private boolean validateFieldValue(String fieldKey, String fieldValue){
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/workflow/service/UserBulkUploadService.java]: The approach in this method is performance consuming... it is converting list of string to comma separated string - everytime when the field validated, reads the string and creates a list by separating with comma -- which is unnecessary.
We have List of String value master_data -- we should store the same in the Redis -- refer hset / hget for the same.
Store list of string in Redis against the key -- just check how to verify the redis contains value for the key in it's list.

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

File: src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImpl.java [hunk 3]
```diff
@@ -1028,7 +1028,6 @@ public SBApiResponse workflowBulkUpdateTransition(String userAuthToken, Multipar
 			uploadedFileDetails.put(Constants.DATE_CREATED_ON, new Timestamp(System.currentTimeMillis()));
 			uploadedFileDetails.put(Constants.STATUS, Constants.INITIATED_CAPITAL);
 			uploadedFileDetails.put(Constants.COMMENT, "");
-			uploadedFileDetails.put(Constants.CREATED_BY, userId);
```
Comment (ravisaurav-tarento) [inline src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImpl.java:1031]: This line is a duplicate code.

File: src/main/java/org/sunbird/workflow/config/Configuration.java (hunk 1 above)
Comment (ravisaurav-tarento) [reply src/main/java/org/sunbird/workflow/config/Configuration.java]: This has been implemented as suggested.

File: src/main/java/org/sunbird/workflow/service/UserBulkUploadService.java (hunk 2 above)
Comment (ravisaurav-tarento) [reply src/main/java/org/sunbird/workflow/service/UserBulkUploadService.java]: This has been implemented as suggested.

**Pull request merged**